src/server/command_registry.cc

```diff
@@ -54,12 +55,19 @@ bool CommandId::IsTransactional() const {
 }
 
 void CommandId::Invoke(CmdArgList args, ConnectionContext* cntx) const {
-  uint64_t before = absl::GetCurrentTimeNanos();
-  handler_(std::move(args), cntx);
-  uint64_t after = absl::GetCurrentTimeNanos();
-  auto& ent = command_stats_[ServerState::tlocal()->thread_index()];
+  ServerState* ss = ServerState::tlocal();
+  int64_t before = absl::GetCurrentTimeNanos();
+  handler_(args, cntx);
+  int64_t after = absl::GetCurrentTimeNanos();
+  int64_t execution_time_micro_s = (after - before) / 1000;
+  auto& ent = command_stats_[ss->thread_index()];
+  if (execution_time_micro_s > ss->log_slower_than_usec) {
+    const auto* conn = cntx->conn();
+    ss->GetSlowLog().Add(name(), args, conn->GetName(), conn->RemoteEndpointStr(),
```

Comment:
We will add here blocking commands that are blocked on a key.. is this the desired behaviour? what does redis do?

Reply:
Redis does not add blocking commands

Reply:
@adiholden is it sufficient if https://github.com/dragonflydb/dragonfly/blob/e6b8cd1d76d7dc128a24313649fbda7eb9c6bd5c/src/server/command_registry.h#L35 are commands discarded?

Reply:
So I think we can add the blocking command to slow log just need to make sure we substruct the time it is waiting on watched key. You can now discard blocking commands and add a TODO comment here. If you wish to implement this as well we will be happy, only making it in a different PR after this one gets merged so it will be easier to review